engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -414,6 +503,6 @@ public void applyOutputShift(@NotNull final RowSet subRowSetToShift, final long
     @Override
     @NotNull
     protected String[] getInputColumnNames() {
-        return inputColumnNames;
+        return ArrayUtils.addAll(inputNonKeyColumnNames);
```

Comment:
I'm unsure about this. I guess a row that moves from one bucket to another is always modified, so we don't need to include the keys here?

Reply:
We handle key modifies as remove (from old bucket) + add (to new bucket).